### ISSUE TITLE
spring-petclinic is not compatible with vaadin 24.6.0.beta3

### ISSUE THREAD

**Issue by ZheSun88** [vaadin/flow#20621]
### Description of the bug

when build the `spring-petclinic`, i got this error

```
Description:

The bean 'endpointObjectMapper', defined in class path resource [org/springframework/boot/actuate/autoconfigure/endpoint/jackson/JacksonEndpointAutoConfiguration.class], could not be registered. A bean with that name has already been defined in class path resource [com/vaadin/hilla/EndpointControllerConfiguration.class] and overriding is disabled.

```

### Expected behavior

project builds ok

### Minimal reproducible example

- git clone git@github.com:spring-petclinic/spring-petclinic-vaadin-flow.git
- update vaadin.version to 24.6.0.beta3
- add vaadin prerelease repo 
  - <repository><id>v</id><url>https://maven.vaadin.com/vaadin-prereleases</url></repository>
  - <pluginRepository><id>v</id><url>https://maven.vaadin.com/vaadin-prereleases</url></pluginRepository>
- ./mvnw spring-boot:run

### Versions

- Vaadin / Flow version: 24.6.0.beta3/24.6.0.beta7
- Java version: 17 (should not matter)

**Comment by ZheSun88** [comment]
one guess is about the version mismatch on jackson.version between
flow 24.6 (jackson 2.18)
spring 3.4 (jackson 2.18)
hilla 24.6 (jackson 2.17)

but hilla cannot be upgraded as swagger-core is not compatible with latest jackson https://github.com/swagger-api/swagger-core/issues/4755

**Comment by taefi** [comment]
No, this is related to some bean that is introduced recently, we should rename that in Hilla.
